File: 정윤/10828.py
#20220415 10828.py

import sys
input = sys.stdin.readline

n = int(input()) 
stack = []

def push(x):
    stack.append(x)

def pop():
    if(len(stack)==0):
        print(-1)
    else:
        print(stack.pop())

def size():
    print(len(stack))

def empty():
    if(len(stack)==0):
        print(1)
    else:
        print(0)

def top():
    if(len(stack)==0):
        print(-1)
    else:
        print(stack[-1])

for i in range(n):
    command = input().split()

    if(command[0]=='push'):
        push(command[1])
    if(command[0]=='pop'):
        pop()
    if(command[0]=='size'):
        size()
    if(command[0]=='empty'):
        empty()
    if(command[0]=='top'):
        top()

File: wonyoung/data_structer/10828.java
// 220415 10828 스택
package wonyoung.data_structer;

import java.io.*;
import java.util.*;

class Main {
    public static Stack<Integer> stack = new Stack<>();

    public static void main(String[] args) throws Exception {
        // for coding
        System.setIn(new FileInputStream("./input/input_10828.txt"));
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        StringTokenizer st;

        int N = Integer.parseInt(br.readLine());

        for (int i = 0; i < N; i++) {
            st = new StringTokenizer(br.readLine(), " ", false);

            // recognize command
            String cmd = st.nextToken();
            if (cmd.equals("push")) {
                int token = Integer.parseInt(st.nextToken());
                stack.push(token);
            } else if (cmd.equals("top")) {
                if (stack.size() == 0)
                    bw.write("-1\n");
                else
                    bw.write(stack.peek() + "\n");
            } else if (cmd.equals("size")) {
                bw.write(stack.size() + "\n");
            } else if (cmd.equals("pop")) {
                if (stack.size() == 0)
                    bw.write("-1\n");
                else
                    bw.write(stack.pop() + "\n");
            } else if (cmd.equals("empty")) {
                bw.write(stack.empty() ? "1" : "0");
                bw.write("\n");
            }
        }

        bw.close();
    }
}